client/src/main/kotlin/io/spine/examples/shareaware/client/Client.kt

```diff
@@ -0,0 +1,237 @@
+     */
+    public fun <S : EntityState> subscribeToEntity(type: Class<S>, id: Message, observer: (S) -> Unit) {
+        clientRequest()
+            .subscribeTo(type)
+            .byId(id)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Subscribes to the events of the provided type.
+     *
+     * @param type type of the event to subscribe on
+     * @param filter filter by which arrived events of the provided type will be filtered
+     * @param observer callback function that will be triggered when the event arrives
+     */
+    public fun <E : EventMessage> subscribeToEvent(
+        type: Class<E>,
+        filter: EventFilter,
+        observer: (E) -> Unit
+    ) {
+        clientRequest()
+            .subscribeToEvent(type)
+            .where(filter)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Retrieves the entity with a provided type and ID.
+     *
+     * @param type type of the entity to be retrieved
+     * @param id entity ID by which the query result will be filtered
+     */
+    public fun <E : EntityState> readEntity(type: Class<E>, id: Message): E? {
+        val entities = clientRequest()
+            .select(type)
+            .byId(id)
+            .run()
+        if (entities.isEmpty()) {
+            return null
+        }
+        return entities[0]
+    }
+
+    /**
+     * Returns the ID of the authenticated user if it exists.
+     *
+     * @throws IllegalStateException when the authenticated user is not configured for the client.
+     */
+    public fun authenticatedUser(): UserId {
+        if (::user.isInitialized) {
+            return user
+        }
+        throw Exceptions.newIllegalStateException(
+            "User has not been authenticated in the system."
+        )
+    }
+
+    /**
+     * Returns the ID of the user's wallet if it exists.
+     *
+     * @throws IllegalStateException when the user's wallet does not exist
+     * because the authenticated user is not configured to the client.
+     */
+    public fun wallet(): WalletId {
+        if (::walletId.isInitialized) {
+            return walletId
+        }
+        throw Exceptions.newIllegalStateException(
+            "There is no user's wallet ID because " +
+                    "the user has not been authenticated in the system."
+        )
+    }
+
+    /**
+     * Returns the `CreateWallet` command.
+     *
+     * @param wallet the ID of the wallet to create
+     */
+    private fun createWallet(wallet: WalletId): CreateWallet {
+        return CreateWallet
+            .newBuilder()
+            .setWallet(wallet)
+            .vBuild()
+    }
+
+    /**
+     * Converts the `UUID` to the `UserId`.
+     */
+    private fun UUID.toUserId(): UserId {
+        return UserId
+            .newBuilder()
+            .setValue(this.toString())
+            .vBuild()
+    }
+
+    /**
+     * Converts the `UserId` to the `WalletId`.
```

Comment:
Please see my comment above on this matter.

We cannot convert one thing into another. They are not equivalent. We can just create something new on top of something that we posess.